### PR TITLE
feat: order introductions contact list by first and last name

### PR DESCRIPTION
**Pull request by iloveitaly**
It's feels much more natural to me to order the introductions list by first and last name.

#### Before submitting the PR
- [x] Read the [CONTRIBUTING document](https://github.com/monicahq/monica/blob/master/CONTRIBUTING.md) before submitting your PR.
- [x] If the PR is related to an issue or fix one, don't forget to indicate it.
- [x] Create your PR as draft if it is not final yet. Mark it as ready... when it’s ready. Otherwise the PR will be considered complete and rejected if it's not working.

### General checks
- [x] Make sure that the change you propose is the smallest possible.
- [x] The name of the PR should follow the [conventional commits guideline](https://github.com/monicahq/monica/blob/master/docs/contribute/readme.md#conventional-commits) that the project follows.

#### Other tasks
- [x] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG.md) entry added, if necessary, under `UNRELEASED`.
- [x] [CONTRIBUTORS](https://github.com/monicahq/monica/blob/master/CONTRIBUTORS) entry added, if necessary.
- [x] If it's relevant and worth mentioning, create a changelog entry for this change. The changelog entry will appear inside the UI for all users to see. To know if your change is worth the creation of a changelog entry, [read the documentation](https://github.com/monicahq/monica/blob/master/docs/administrators/tips.md#when-is-it-relevant-to-create-a-changelog-entry).
- [x] Don't forget to [ask for a free account](mailto:regis@monicahq.com) on https://monicahq.com as anyone who contributes can request a free account.

### CODE REVIEW

**Comment by djaiss** [comment]
Hi @iloveitaly and thanks for the PR.

To you it might feel more natural, but in other cultures, it might not.

This is why we introduced the column called `name_order` in the `users` table that define how people want their contact to be sorted.

I think we should use this in this case, what do you think @asbiin ?

**Comment by iloveitaly** [comment]
Ah, great, didn't realize there was an option for that. It sounds like we should add a [switch similar to this one](https://github.com/monicahq/monica/blob/708eb6090ebe8ef71546e99621718c5786712dee/app/Helpers/FormHelper.php#L16) to add 2-3 `orderBy`s to the query. I'm not that familiar with the codebase—where would be the best place for this helper?

**Comment by djaiss** [comment]
> Ah, great, didn't realize there was an option for that. It sounds like we should add a [switch similar to this one](https://github.com/monicahq/monica/blob/708eb6090ebe8ef71546e99621718c5786712dee/app/Helpers/FormHelper.php#L16) to add 2-3 `orderBy`s to the query. I'm not that familiar with the codebase—where would be the best place for this helper?

Good question.
We might need to use this in other places too. So I would perhaps create a `ContactHelper` with the right method name in it, like `getNameInUserOrder` or something like this (yes, naming is hard).

File: app/Helpers/ContactHelper.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+use App\Models\User\User;
+
+class ContactHelper
+{
+    /**
+     * Get the name order that will be used when rendered the Add/Edit forms
+     * about contacts.
+     *
+     * @param Builder $query
+     * @param User $user
```
Comment (asbiin) [inline app/Helpers/ContactHelper.php:14]:
```suggestion
     * @param User $user
     * @param \Illuminate\Database\Eloquent\Builder $query
```

File: app/Helpers/ContactHelper.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Models\User\User;
```
Comment (asbiin) [inline app/Helpers/ContactHelper.php:5]:
```suggestion
use App\Models\User\User;
use Illuminate\Database\Eloquent\Builder;
```

File: app/Helpers/ContactHelper.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+{
+    /**
+     * Get the name order that will be used when rendered the Add/Edit forms
+     * about contacts.
+     *
+     * @param Builder $query
+     * @param User $user
+     * @return string
+     */
+    public static function orderContactQueryByUserPreference(User $user, $query)
```
Comment (asbiin) [inline app/Helpers/ContactHelper.php:17]:
```suggestion
     * @return Builder
     */
    public static function orderContactQueryByUserPreference(User $user, $query): Builder
```

Comment (iloveitaly) [reply app/Helpers/ContactHelper.php]: I couldn't get the builder return type to work for me. Looks like it's failing the build. Let me know if you want to me to make any other changes here to get this merged. Feel free to modify this branch as much as you'd like!

**Comment by iloveitaly** [comment]
@asbiin Any thoughts on how to fix the typing error? `Builder` as a param + return type seems to return an error since `HasMany` is passed in.

File: app/Helpers/ContactHelper.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+{
+    /**
+     * Get the name order that will be used when rendered the Add/Edit forms
+     * about contacts.
+     *
+     * @param User $user
+     * @param Builder $query
+     * @return Builder
+     */
+    public static function orderContactQueryByUserPreference(User $user, $query): Builder
```
Comment (asbiin) [inline app/Helpers/ContactHelper.php:18]: In fact if you want to use Builder here, it's better to use a [scope](https://laravel.com/docs/8.x/eloquent#local-scopes)
See https://github.com/monicahq/monica/blob/master/app/Traits/Searchable.php

**Review by asbiin — CHANGES_REQUESTED** [review]
phpstan is complaining, and I think it would be better to use a scope

**Pull request merged**